include/PathSimulator.hpp

```diff
@@ -0,0 +1,190 @@
+#ifndef DDSIM_PATHSIMULATOR_HPP
+#define DDSIM_PATHSIMULATOR_HPP
+
+#include "CircuitOptimizer.hpp"
+#include "CircuitSimulator.hpp"
+#include "nlohmann/json.hpp"
+
+#include <future>
+#include <limits>
+#include <taskflow/taskflow.hpp>
+#include <thread>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <variant>
+#include <vector>
+
+class PathSimulator: public CircuitSimulator {
+public:
+    struct SimulationPath {
+        struct Step {
+            std::size_t                         id;
+            std::vector<std::size_t>            operations;
+            std::size_t                         parent;
+            std::pair<std::size_t, std::size_t> children;
+
+            explicit Step(std::size_t id, std::vector<std::size_t> operations = {}, std::size_t parent = UNKNOWN, std::pair<std::size_t, std::size_t> children = {UNKNOWN, UNKNOWN}):
+                id(id), operations(std::move(operations)), parent(parent), children(std::move(children)){};
+
+            static constexpr size_t UNKNOWN = std::numeric_limits<size_t>::max();
+        };
+
+        using Components = std::vector<std::pair<std::size_t, std::size_t>>;
+        using Steps      = std::vector<Step>;
+
+        SimulationPath() = default;
+        SimulationPath(std::size_t nleaves, Components components, const qc::QuantumComputation* qc, bool assumeCorrectOrder = false);
+
+        Components                    components{};
+        Steps                         steps{};
+        std::size_t                   nleaves{};
+        const qc::QuantumComputation* qc{};
+    };
+
+    struct Configuration {
+        // Add new strategies here
+        enum class Mode {
+            Sequential,
+            PairwiseRecursiveGrouping,
+            BracketGrouping,
+            Alternating,
+            Cotengra
+        };
+
+        // mode to use
+        Mode mode;
+
+        // settings for the bracket size
+        std::size_t bracketSize;
+        // settings for the alternating mode
+        std::size_t alternatingStart;
+
+        // random seed
+        std::size_t seed;
```

Comment:
Could these be `const`?

Reply:
i don't think so. they might be modified after construction. quite frequently you just default construct a configuration object and then change a single option. wouldn't be possible with them being const.

Reply:
Okay.